packages/react/src/components/Geo/LocationSearch/index.tsx

```diff
@@ -15,8 +16,35 @@ const LOCATION_SEARCH_OPTIONS = {
 
 const LOCATION_SEARCH_CONTAINER = 'geocoder-container';
 
+interface LocationSearchProps extends MaplibreGeocoderOptions {
+  /**
+   * Emitted when the input is cleared
+   */
+  onClear: () => void;
+  /**
+   * Emitted when the geocoder is looking up a query
+   */
+  onLoading: (query) => void;
+  /**
+   * Fired when the geocoder returns a response
+   */
+  onResults: (results) => void;
+  /**
+   * Fired when input is set
+   */
+  onResult: (result) => void;
+  /**
+   * Emitted on error as string
+   */
+  onError: (error) => void;
+}
+
+type eventTypes = 'clear' | 'loading' | 'result' | 'results' | 'error';
+
 type AmplifyLocationSearch = IControl & {
   addTo: (container: string) => void;
+  on: (eventType: eventTypes, callback: (event) => void) => void;
```

Comment:
Since `onClear` does not emit an event to its callback, should we should consider using a union type here?

Reply:
Not sure I'm following on this one.